doc/MiniLanguageTutorial.md

```diff
@@ -112,31 +139,56 @@ Mini has the following types:
 
 > an untagged union type that can hold values from any of *type1*, *type2*, etc. There must be explicit casts to convert to and from its component types. This is done via `newunion` and `unioncast`.
 
-`struct` { *name1: type1 , name2 : type2 , ...* }
+`struct` `{` *name1* `:` *type1* `,` *name2* `:` *type2* `,` ... `}`
 
 > a struct with one or more named, typed fields (a compound type)
 
 `option`< *type* >
 
-> either the contained type or "None<*type*>", must be unwrapped to compare with inner type
+> Either `Some(`*value*`)` or `None<`*type*`>`, where *value* represents an arbitrary value of type *type*. 
+> The specific variant being used can be determined by `if let` and `?`.
+> The shortcut `None` can be used to create a value of `None<every>`, which is assignable to all `None<`T`>` for any type T.
+> The term "inner type" refers to the type specified by *type*.
+
+[ `view` | `write` | `public` ]* `func` ( *type1, type2, ...*) (-> *returntype*)?
+
+[ `view` | `write` ] `closure` `(` *type*, *type*, ... `)` (`->` *returntype*`)?
 
-[`impure`] `func` ( *type1, type2, ...*) [-> *returntype*]
+> A reference to a function, while `public` can be included a type declaration, it has no effect on the internal type.
 
-> a reference to a function
+*ident*
+
+> A nominal type or a type variable. It is only treated as a type variable in the context of a generic type or function,
+> and only if that identifier is listed as one of the type variables of that generic type or function. 
+> Otherwise, it is treated as a nominal type, if that nominal type is not defined in the file or is not imported with a use statement, 
+> it is not a valid type and will cause an error.
+> The representation of a nominal type is the type on the left side of the type declaration with the same name as *ident*.
```

Comment:
I cleared up that we are defining a new term rather than referencing a existing definition, and changed left to right, which was a mistake. I am going to leave this open so you can check if this is understandable.